test/common.js

```diff
@@ -78,78 +81,109 @@ if (testing('int')) {
+  }
+
+  const currentName = newName();
+  module.exports.currentName = currentName;
+
+  module.exports.api = new Core({
     url: url,
     ca: ca,
     cert: cert,
     key: key,
     version: process.env.VERSION || 'v1',
-    namespace: defaultName
+    namespace: currentName
   });
 
-  extensions = new Extensions({
+  module.exports.extensions = new Extensions({
     url: url,
     ca: ca,
     cert: cert,
     key: key,
     version: process.env.VERSION || 'v1beta1',
-    namespace: defaultName
+    namespace: currentName
   });
 
-  apiGroup = new Api({
+  module.exports.apiGroup = new Api({
     url: url,
     ca: ca,
     cert: cert,
     key: key,
-    namespace: defaultName
+    namespace: currentName
   });
 
-  api.wipe = function (cb) {
+  module.exports.api.ns.post({
+    body: {
+      kind: 'Namespace',
+      metadata: {
+        name: currentName
+      }
+    }
+  }, err => {
+    if (err) return cb(err);
     const times = Math.ceil(defaultTimeout / 1000);
     const interval = 1000;
     async.retry({ times: times, interval: interval }, next => {
-      this.ns.delete({ name: defaultName }, () => {
-        this.ns.get({ name: defaultName }, err => {
-          if (!err) return next(new Error(`${ defaultName} still exists`));
-          if (err.code !== 404) return next(err);
-          this.ns.post({ body: {
-            kind: 'Namespace',
-            metadata: {
-              name: defaultName
-            }
-          }}, next);
-        });
-      });
-    }, cb);
-  }
-  api.wipe = api.wipe.bind(api);
-} else {
-  api = new Core({
+      module.exports.api.ns.serviceaccounts.get('default', (saErr, sa) => {
+        if (saErr) return next(saErr);
+        if (!sa.secrets) {
+          return next(new Error('Waiting for servicesaccount secrets'));
+        }
+        cb();
+      })
+    });
+  });
+}
+
+function changeNameUnit() {
+  const currentName = newName();
+  module.exports.currentName = currentName;
+
+  module.exports.api = new Core({
     url: 'http://mock.kube.api',
     version: process.env.VERSION || 'v1',
-    namespace: defaultName
+    namespace: currentName
   });
 
-  extensions = new Extensions({
+  module.exports.extensions = new Extensions({
     url: 'http://mock.kube.api',
     version: process.env.VERSION || 'v1beta1',
-    namespace: defaultName
+    namespace: currentName
   });
 
-  apiGroup = new Api({
+  module.exports.apiGroup = new Api({
     url: 'http://mock.kube.api',
-    namespace: defaultName
+    namespace: currentName
   });
+}
 
-  api.wipe = () => {
-    throw new Error("Don't call wipe during unit tests");
+function changeName(cb) {
+  if (testing('int')) return changeNameInt(cb);
+
+  throw new Error('Do not call changeName during unit tests');
+}
+
+if (!testing('int')) {
+  changeNameUnit();
+}
+
+//
+// Do our best to cleanup namespaces, but it's hard to ensure we don't leak.
+//
+function atexit() {
+  if (testing('int') && module.exports.currentName) {
```

Comment:
yea this probably wont work but not the biggest deal. Maybe we should have an `after` that gets run that clears all namespaces when we are `testing('int')`? I think thats the cleanest way to try and do this for real.